storage/app/api/v1/attachments/controller.py

```diff
@@ -1,27 +1,22 @@
-import os
 import io
 import sys
-import magic
-import shutil
 import uuid
 import logging
 import zipfile
-from os import path, replace
-from pathlib import Path
 from fastapi.responses import StreamingResponse, Response
-from tempfile import NamedTemporaryFile
 from app.schema import Attachment, Attachments
 from fastapi import UploadFile, HTTPException, Depends
-from starlette.status import HTTP_500_INTERNAL_SERVER_ERROR, HTTP_415_UNSUPPORTED_MEDIA_TYPE
-from app.core.minio import s3_upload_file, s3_delete_file, s3_list_directory, s3_get_file, s3_upload_raw_file
-from app.utils import get_file_ext, get_file_name, generate_file_name
-from app.config import ATTACHMENTS_BUCKET_NAME, ALLOWED_FILE_EXTENSIONS
+from starlette.status import HTTP_500_INTERNAL_SERVER_ERROR
+from app.core.minio import s3_delete_file, s3_get_file, s3_get_file_stats, s3_upload_raw_file
+from app.config import ATTACHMENTS_BUCKET_NAME
 
 logger = logging.getLogger("attachments")
 
 
 def s3_response_iterator(s3_response, chunk_size=2048):
     while 1:
+        # this is not very useful as long as the MinIO client is synchronous
+        # The read() call will need to be awaited when using an async client.
```

Comment:
I've been warned by other users of the python MinIO clients in the gov that its API is synchronous, which means we'll run into a bottleneck pretty quickly when reading multiple files concurrently. This can be addressed with https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/503